src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -415,7 +417,7 @@ internal TexlBinding ApplyBindingInternal()
                     }
                 }
 
-                if (this.ReturnType != null && this._expectedReturnType != null)
+                if (!_allowCoerceToType && this.ReturnType != null && this._expectedReturnType != null)

```

Comment:
>_allowCoerceToType [](http://example.com/codeflow?start=21&length=18)

                if (this.ReturnType != null && this._expectedReturnType != null)
                {
                    bool Ok;
                    if (_allowCoerce)
                    {
                        if (this._expectedReturnType != FormulaType.String)
                        {
                            throw new NotImplementedException();
                        }
                        // Use StringValue.AllowedList
                    } else
                    {
                        // Strict 
                        ok = this._expectedReturnType == this.ReturnType;
                    }

                    if (!ok)

Reply:
updated logic